src/forces.f90

```diff
@@ -256,267 +268,626 @@ subroutine force(ux1,uy1,ep1)
 
     call derx (ta1,ux1,di1,sx,ffx,fsx,fwx,xsize(1),xsize(2),xsize(3),0,ubcx)    ! dudx
     call derx (tb1,uy1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcy) ! dvdx
+    call derx (tc1,uz1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcz) ! dwdx
```

Comment:
Can we use te1 here ? It is best to have a PR that keep the 2D force calculation unchanged.

This operation is not needed when i2dsim = 1 (default). Move all the 3D-specific operations together and perform them only if needed.